File: src/main/java/infinitespire/cards/black/DeathsTouch.java
package infinitespire.cards.black;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.actions.common.DamageAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import com.megacrit.cardcrawl.powers.TheBombPower;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.BlackCard;

public class DeathsTouch extends BlackCard {

	public static final String ID = InfiniteSpire.createID("DeathsTouch");

	private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
	private static final String NAME = cardStrings.NAME;
	private static final String IMG = "img/infinitespire/cards/deathstouch.png";
	private static final int COST = 3;
	private static final String DESCRIPTION = cardStrings.DESCRIPTION;
	private static final CardType TYPE = CardType.ATTACK;
	private static final CardTarget TARGET = CardTarget.ENEMY;
	
	private static final int DAMAGE = 25;
	
	public DeathsTouch() {
		super(ID, NAME, IMG, COST, DESCRIPTION, TYPE, TARGET);
		this.baseDamage = DAMAGE;
	}
	
	@Override
	public AbstractCard makeCopy() {
		return new DeathsTouch();
	}

	@Override
	public void upgrade() {
		if(!upgraded) {
			this.upgradeName();
			this.upgradeDamage(10);
		}
	}

	@Override
	public void useWithEffect(AbstractPlayer p, AbstractMonster m) {
		AbstractDungeon.actionManager.addToBottom(new DamageAction(m, new DamageInfo(p, this.damage), AbstractGameAction.AttackEffect.FIRE));
		AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(p, p, new TheBombPower(p, 2, this.damage)));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/cards/black/DeathsTouch.kt
package com.blanktheevil.infinitespire.cards.black

import com.blanktheevil.infinitespire.cards.utils.CardBuilder
import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.actions.common.DamageAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.powers.TheBombPower

class DeathsTouch : BlackCard(BUILDER) {
  companion object {
    val ID = "DeathsTouch".makeID()
    private const val DAMAGE = 25
    private const val UPG_DAMAGE = 10
    private val BUILDER = CardBuilder(ID)
      .img("deathstouch.png")
      .cost(3)
      .attack()
      .enemy()
      .init {
        baseDamage = DAMAGE
      }
      .upgr {
        upgradeDamage(UPG_DAMAGE)
      }
      .use { player, monster ->
        addToBot(DamageAction(
          monster,
          DamageInfo(player, damage),
          AbstractGameAction.AttackEffect.FIRE
        ))
        player?.applyPower(
          TheBombPower(player, 2, this.damage)
        )
      }
  }
}